src/main/java/ro/developmentfactory/thedrones/service/DronePositionServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+package ro.developmentfactory.thedrones.service;
+
+import jakarta.persistence.EntityNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ro.developmentfactory.thedrones.controller.dto.DroneStatusResponse;
+import ro.developmentfactory.thedrones.repository.DroneRepository;
+import ro.developmentfactory.thedrones.repository.DroneStatusRepository;
+import ro.developmentfactory.thedrones.repository.entity.Drone;
+import ro.developmentfactory.thedrones.repository.entity.DroneStatus;
+
+import java.util.UUID;
+
+@Service
+public class DronePositionServiceImpl implements DronePositionService {
+
+    private final DroneStatusRepository droneStatusRepository;
+    private final DroneRepository droneRepository;
+    private static final Logger log = LoggerFactory.getLogger(DronePositionServiceImpl.class);
+
+    public DronePositionServiceImpl(DroneStatusRepository droneStatusRepository, DroneRepository droneRepository) {
+        this.droneStatusRepository = droneStatusRepository;
+        this.droneRepository = droneRepository;
+    }
+
+    @Override
+    @Transactional
+    public DroneStatusResponse turnDroneLeft(DroneStatusResponse droneStatusResponse) {
+        log.debug("Turning drone left :{}",droneStatusResponse);
+        UUID idDrone = droneStatusResponse.getIdDrone();
+
+        Drone drone = droneRepository.findById(idDrone)
+                .orElseThrow(() -> new EntityNotFoundException("Drone with id " + idDrone + " not found"));
+
+        DroneStatus currentStatus = drone.getDroneStatusList().getFirst();
+        currentStatus.setFacingDirection(currentStatus.getFacingDirection().turnLeft());
+
+        droneStatusRepository.save(currentStatus);
+       return convertToResponse(currentStatus);
+    }
+
+    @Override
+    @Transactional
+    public DroneStatusResponse turnDroneRight(DroneStatusResponse droneStatusResponse) {
+        log.debug("Turning drone right :{}",droneStatusResponse);
+        UUID idDrone = droneStatusResponse.getIdDrone();
+
+        Drone drone = droneRepository.findById(idDrone)
+                .orElseThrow(() -> new EntityNotFoundException("Drone with id " + idDrone + " not found"));
+
+        DroneStatus currentStatus = drone.getDroneStatusList().getFirst();
+        currentStatus.setFacingDirection(currentStatus.getFacingDirection().turnRight());
+
+        droneStatusRepository.save(currentStatus);
+        return convertToResponse(currentStatus);
+    }
+
+    @Override
+    @Transactional
+    public DroneStatusResponse moveForward(DroneStatusResponse droneStatusResponse) {
+        log.debug("Moving forward for drone status :{}",droneStatusResponse);
+
+        DroneStatus droneStatus = droneStatusRepository.findById(droneStatusResponse.getIdDroneStatus())
```

Comment:
I would probably remove `idDroneStatus` from `DroneStatusResponse` response too - for the same reason - API users should not know internal data organization and ids that they don't need to operate your API